lib/spack/spack/audit.py

```diff
@@ -275,15 +283,40 @@ def _search_duplicate_specs_in_externals(error_cls):
 )
 
 
+@package_github_patch_url
+def _check_patch_urls(pkgs, error_cls):
+    """Ensure that patches fetched from GitHub have stable sha256 hashes."""
+    github_patch_url_re = (
+        r"^https?://github\.com/.+/.+/(?:commit|pull)/[a-fA-F0-9]*.(?:patch|diff)"
+    )
```

Comment:
Minor point, but we can pre-compile the regex out of the loop.

Reply:
Python looks up already compiled regexes in an internal dictionary, so I’m not too worried about this.